actors/src/main/scala-2/zio/actors/ActorSystem.scala

```diff
@@ -0,0 +1,75 @@
+package zio.actors
+
+import zio.actors.ActorSystemUtils._
+import zio.actors.ActorsConfig._
+import zio.{ Ref, Task, UIO, ZIO }
+
+import java.io._
+
+/**
+ * Object providing constructor for Actor System with optional remoting module.
+ */
+object ActorSystem {
+
+  /**
+   * Constructor for Actor System
+   *
+   * @param sysName
+   *   \- Identifier for Actor System
+   * @param configFile
+   *   \- Optional configuration for a remoting internal module. If not provided the actor system will only handle local
+   *   actors in terms of actor selection. When provided - remote messaging and remote actor selection is possible
+   * @return
+   *   instantiated actor system
+   */
+  def apply(sysName: String, configFile: Option[File] = None): Task[ActorSystem] =
+    for {
+      initActorRefMap <- Ref.make(Map.empty[String, Actor[Any]])
+      config          <- retrieveConfig(configFile)
+      remoteConfig    <- retrieveRemoteConfig(sysName, config)
+      actorSystem     <- ZIO.attempt(new ActorSystem(sysName, config, remoteConfig, initActorRefMap, parentActor = None))
+      _               <- ZIO
+                           .succeed(remoteConfig)
+                           .flatMap(_.fold[Task[Unit]](ZIO.unit)(c => actorSystem.receiveLoop(c.addr, c.port)))
+    } yield actorSystem
+}
+
+/**
+ * Type representing running instance of actor system provisioning actor herding, remoting and actor creation and
+ * selection.
+ */
+final class ActorSystem private[actors] (
+  override private[actors] val actorSystemName: String,
+  override private[actors] val config: Option[String],
+  private val remoteConfig: Option[RemoteConfig],
+  private val initActorRefMap: Ref[Map[String, Actor[Any]]],
```

Comment:
If I see it correctly the only version-specific difference is `Actor[Any]` vs `Actor[?]` here. Is Scala 2 not compiling with `?` (or `_`?)?

Reply:
That's correct, I have made a few attempts on scala 2.13 but haven't been able to make it work... Get errors like:
```
private val initActorRefMap: Ref[Map[String, Actor[_]]], _$2 takes no type parameters, expected: 1
                                                   ^
```

I have also tried to make scala 3 compile with `Actor[Any]` but get errors like: 
```
Type argument Any does not have the same kind as its bound [+_$1]
```